.github/workflows/test.yml

```
@@ -0,0 +1,41 @@
+name: Test
+
+on:
+  pull_request:
+    branches:
+      - main
+  schedule:
+    - cron: '0 22 * * 3'
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    name: Test
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version:
+          - '20.19'
+        os:
+          - macos-latest
+          - ubuntu-22.04
```

```suggestion
          - ubuntu-latest
```

If `ubuntu-latest` works here we should use it, a couple of repos had problems with it and the fastest fix was to force `ubuntu-22.04` but ideally we don't pin to that version since it's on the way out.